libglusterfs/src/glusterfs/list.h

```diff
@@ -66,21 +66,31 @@ list_add_order(struct list_head *new, struct list_head *head,
     list_add(new, pos);
 }
 
+/* Delete a list entry by making the prev/next entries
+   point to each other.
+
+   This is only for internal list manipulation where we know
+   the prev/next entries already! */
 static inline void
-list_del(struct list_head *old)
+_list_del(struct list_head *old)
 {
-    old->prev->next = old->next;
     old->next->prev = old->prev;
+    old->prev->next = old->next;
+}
 
-    old->next = (void *)0xbabebabe;
-    old->prev = (void *)0xcafecafe;
+static inline void
+list_del(struct list_head *old)
+{
+    _list_del(old);
+
+    old->next = (void *)0x00100100;
+    old->prev = (void *)0x00200200;
```

Comment:
These values should cause a memory access error if used. The current value seems like a potentially valid address to me that could cause silent memory corruption if used (I prefer a crash in this case). I would use a smaller value (below 0x1000) or a negative value (0xfff...) to be sure. Those addresses should never be valid in user space code.

Reply:
It would also be better to define constants for this.

Reply:
We can put there any value you think make sense - those were copied from the Linux kernel. 0x1 and 0x2 are also good, IMHO.

Reply:
I would use values that are unlikely to be there by chance so that it's easy to identify them when debugging. I think 1 and 2 are too common values.

Latest kernels are using 0xdead000000000000 + some small value (only for x64, for 32 bits they use 0). I think we could use the same value with 0x100 and 0x200 as the small values.

Reply:
Done, I hope.